llvm/runtimes/CMakeLists.txt

```diff
@@ -138,37 +142,44 @@ endfunction()
 # before the just-built compiler can pass the configuration tests.
 get_compiler_rt_path(compiler_rt_path)
 if(compiler_rt_path)
-  if(NOT LLVM_BUILTIN_TARGETS)
+  # If the user did not specify the targets infer them from the runtimes.
+  set(builtin_targets ${LLVM_BUILTIN_TARGETS})
+  if(NOT builtin_targets)
+    if("compiler-rt" IN_LIST LLVM_ENABLE_RUNTIMES)
+      list(APPEND builtin_targets "default")
+    endif()
```

Comment:
I'm not sure that applying DRY here is worth it. I'd rather just duplicate the `builtin_default_target` invocation and remove the string handling for "default".

Reply:
We need the following case to work,
```
   -DLLVM_ENABLE_RUNTIMES=compiler-rt \
   -DRUNTIMES_nvptx64-nvidia-cuda_LLVM_ENABLE_RUNTIMES=compiler-rt               \
   -DRUNTIMES_amdgcn-amd-amdhsa_LLVM_ENABLE_RUNTIMES=compiler-rt                 \
   -DLLVM_RUNTIME_TARGETS="default;amdgcn-amd-amdhsa;nvptx64-nvidia-cuda"
```
Honestly I think we should get rid of the distinction between the "default" target, but that's a much larger refactoring I think @petrhosek wanted to do.

Reply:
The "default" target is also important for how the compiler-rt parts of this are built on Darwin, which is a mess.

Reply:
> I'm not sure that applying DRY here is worth it. I'd rather just duplicate the `builtin_default_target` invocation and remove the string handling for "default".

I have a WIP change that removes the `"default"` altogether since that part is a relic, but it needs some more testing to make sure I don't break any of existing configurations.